jfrog-cli/artifactory/utils/golang/project/dependencies/dependencies.go

```diff
@@ -1,216 +1,108 @@
-
-	if err != nil {
-		return nil, err
-	}
-
-	if zipPath == "" {
-		return nil, nil
-	}
-
-	dep := Dependency{}
-
-	dep.id = strings.Join([]string{dependencyName, version}, ":")
-	dep.version = version
-	dep.zipPath = zipPath
-	dep.modContent, err = ioutil.ReadFile(filepath.Join(cachePath, dependencyName, "@v", version+".mod"))
-	if err != nil {
-		return &dep, errorutils.CheckError(err)
-	}
-
-	// Mod file dependency
-	modDependency := buildinfo.Dependency{Id: dep.id}
-	checksums, err := checksum.Calc(bytes.NewBuffer(dep.modContent))
-	if err != nil {
-		return &dep, err
-	}
-	modDependency.Checksum = &buildinfo.Checksum{Sha1: checksums[checksum.SHA1], Md5: checksums[checksum.MD5]}
-
-	// Zip file dependency
-	zipDependency := buildinfo.Dependency{Id: dep.id}
-	fileDetails, err := fileutils.GetFileDetails(dep.zipPath)
-	if err != nil {
-		return &dep, err
-	}
-	zipDependency.Checksum = &buildinfo.Checksum{Sha1: fileDetails.Checksum.Sha1, Md5: fileDetails.Checksum.Md5}
-
-	dep.buildInfoDependencies = append(dep.buildInfoDependencies, modDependency, zipDependency)
-	return &dep, nil
+// Init the dependency information if needed.
+func (dependencyPackage *Package) Init() error {
+	return nil
 }
 
-// Returns the path to the package zip file if exists.
-func getPackageZipLocation(cachePath, dependencyName, version string) (string, error) {
-	zipPath, err := getPackagePathIfExists(cachePath, dependencyName, version)
-	if err != nil {
-		return "", err
-	}
-
-	if zipPath != "" {
-		return zipPath, nil
-	}
-
-	zipPath, err = getPackagePathIfExists(filepath.Dir(cachePath), dependencyName, version)
-
-	if err != nil {
-		return "", err
-	}
-
-	return zipPath, nil
+func (dependencyPackage *Package) PopulateModIfNeededAndPublish(targetRepo string, cache *golangutil.DynamicCache, details *config.ArtifactoryDetails) error {
+	return dependencyPackage.prepareAndPublish(targetRepo, cache, details)
 }
-
-// Validates if the package zip file exists.
-func getPackagePathIfExists(cachePath, dependencyName, version string) (zipPath string, err error) {
-	zipPath = filepath.Join(cachePath, dependencyName, "@v", version+".zip")
-	fileExists, err := fileutils.IsFileExists(zipPath, false)
+// Prepare for publishing and publish the dependency to Artifactory
+func (dependencyPackage *Package) prepareAndPublish(targetRepo string, cache *golangutil.DynamicCache, details *config.ArtifactoryDetails) error {
+	serviceManager, err := utils.CreateServiceManager(details, false)
 	if err != nil {
-		log.Warn(fmt.Sprintf("Could not find zip binary for dependency '%s' at %s.", dependencyName, zipPath))
-		return "", err
+		cache.IncreaseFailures()
+		return err
 	}
-	// Zip binary does not exist, so we skip it by returning a nil dependency.
-	if !fileExists {
-		log.Debug("The following file is missing:", zipPath)
-		return "", nil
+	totalOutOf := fmt.Sprintf("%d/%d", cache.GetSuccess()+1, cache.GetTotal())
```

Comment:
I suggest renaming *totalOutOf* to *successOutOfTotal*

Reply:
Done